mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -1008,10 +1008,10 @@ RocmlirSplitKSelectionLikelihood isSplitKFaster(int64_t gDim, int64_t mDim,
 }
 
 bool isModuleFusible(ModuleOp module, StringRef perfConfig) {
-  if (!rock::isSplitKRequested(module, perfConfig)) {
-    return true;
-  }
-  return succeeded(rock::testFusionLegality(module));
+  bool fusible = succeeded(rock::testFusionLegalityReduce(module));
+  if (!rock::isSplitKRequested(module, perfConfig))
+    return fusible;
+  return fusible && succeeded(rock::testFusionLegality(module));
```

Comment:
ideally speaking `testFusionLegalityReduce` should be subset of `testFusionLegality()` therefore we shouldn't really need to check both